### PR TITLE
systemd::unit_file: remove hasrestart/hasstatus params

### PR DESCRIPTION
**Pull request by bastelfreak**
This removes two parameters from systemd::unit_file. In release 3.8.0 on 2022-03-02 we introcuced the deprecation warnings. We should merge this PR somewhere in the future when we've other breaking changes. Do not merge at the moment!

### CODE REVIEW

**Review by simondeziel — APPROVED** [review]

File: manifests/unit_file.pp [hunk 1]
```diff
@@ -81,23 +75,13 @@
   Optional[Variant[Boolean, Enum['mask']]] $enable    = undef,
   Optional[Boolean]                        $active    = undef,
   Optional[String]                         $restart   = undef,
```
[the remark this answers is not in the document]
Comment (ekohl) [reply manifests/unit_file.pp:77]: This is the `restart` parameter on `service`, which I think can still be useful. For example, one could set `/usr/bin/systemctl reload-or-restart $unit`.

If you're talking about the one in `service_limits`, then I'd agree. IMHO that's a bad pattern.
https://github.com/voxpupuli/puppet-systemd/blob/d9d99c3d660027d8df5551792411a16b79370933/manifests/service_limits.pp#L38

[the remark this answers is not in the document]
Comment (traylenator) [reply manifests/unit_file.pp:77]:
```puppet
systemd::unit_file { "my.service":
  ....
  service_parameters => {
    'restart' => "/usr/bin/systemctl reload-or-restart $unit"
  }
}
```
is enough. .. Or is intention to just not add every single option that could possibly go to service - also makes sense.

[the remark this answers is not in the document]
Comment (ekohl) [reply manifests/unit_file.pp:77]: Oh that's an excellent point. I think overriding this is obscure enough that it's valid to not have a dedicated parameter. :+1:

**Review by jhoblitt — APPROVED** [review]
This seems reasonable to me.  I will remove the REFERENCE.md changes from the branch to resolve the merge conflict.

**Comment by jhoblitt** [comment]
Doh, I just noticed this PR is from a fork.

**Pull request merged**